datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/ResolverUtils.java

```diff
@@ -81,20 +83,73 @@ public static Map<String, String> buildFacetFilters(@Nullable List<FacetFilterIn
             if (!validFacetFields.contains(facetFilterInput.getField())) {
                 throw new ValidationException(String.format("Unrecognized facet with name %s provided", facetFilterInput.getField()));
             }
-            facetFilters.put(facetFilterInput.getField(), facetFilterInput.getValue());
+            facetFilters.put(facetFilterInput.getField(), facetFilterInput.getValues().get(0));
         });
 
         return facetFilters;
     }
 
+    // In the case that user sends filters to be or-d together, we need to build a series of conjective criterion
+    // arrays, rather than just one for the AND case.
+    public static ConjunctiveCriterionArray buildConjunctiveCriterionArrayWithOr(
+        final List<Criterion> andCriterions,
+        @Nonnull List<FacetFilterInput> orFilters
+    ) {
+        return new ConjunctiveCriterionArray(orFilters.stream().map(orFilter -> {
+                CriterionArray andCriterionsForOr = new CriterionArray(andCriterions);
+                andCriterionsForOr.add(criterionFromFilter(orFilter));
+
+                return new ConjunctiveCriterion().setAnd(
+                    andCriterionsForOr
+                );
+            }
+        ).collect(Collectors.toList()));
+    }
+
     @Nullable
-    public static Filter buildFilter(@Nullable List<FacetFilterInput> facetFilterInputs) {
-        if (facetFilterInputs == null || facetFilterInputs.isEmpty()) {
+    public static Filter buildFilter(@Nullable List<FacetFilterInput> andFilters, @Nullable List<FacetFilterInput> orFilters) {
+        if ((andFilters == null || andFilters.isEmpty()) && (orFilters == null || orFilters.isEmpty())) {
```

Comment:
can you separate these two conditions out into boolean variables? it's a little hard to read, and you can use the the same variables down below when building the andCriterions and orCriterions.

Maybe `hasAndFilters` and `hasOrFilters`?

Reply:
resolved